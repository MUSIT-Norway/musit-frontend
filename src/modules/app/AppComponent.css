nav.fa {
  font-size: 1em;
  line-height: 20px;
}

.brand {
  position: absolute;
  top: 5px;
  left: 5px;
  display: inline-block;

  width: 20px;
  height: 20px;
  background-size: 80%;
  margin: 0 10px 0 0;
  border-radius: 10px;
}

.appContent {
  margin: 50px 0;
}

.footer {
  text-align: right;
  position: fixed;
  bottom: 0;
  width: 100%;
}

.version {
  font-size: smaller;
  color: lightgray;
}

.version a {
  font-size: smaller;
  text-decoration: none;
  color: lightgray !important;
}

.backgroundUTV {
  height: 50px;
  background-color: #ff6666;
  font-size: smaller;
  color: black;
  margin-bottom: 1;
  padding-right: 10px;
}

.backgroundTEST {
  height: 50px;
  background-color: orange;
  font-size: smaller;
  color: black;
  margin-bottom: 1;
  padding-right: 10px;
}
.footer.well {
  margin-bottom: 0;
}
