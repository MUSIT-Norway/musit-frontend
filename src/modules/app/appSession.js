import { Observable } from 'rxjs';
import { createStore, createAction } from 'react-rxjs/dist/RxStore';
import { simpleGet } from '../../shared/RxAjax';
import Config from '../../config';
import { getAccessToken } from '../../shared/token';
import { emitError } from '../../shared/errors';
import { getDisplayName } from '../../shared/util';
import { I18n } from 'react-i18nify';
import orderBy from 'lodash/orderBy';
import React from 'react';
import PropTypes from 'prop-types';
import isEqualWith from 'lodash/isEqualWith';
import { getLanguage } from './../../shared/language';

export const makeUrlAware = Component => {
  class Wrapper extends React.Component {
    static propTypes = {
      appSession: PropTypes.shape({
        museumId: PropTypes.number.isRequired,
        collectionId: PropTypes.string.isRequired
      }).isRequired,
      params: PropTypes.shape({
        museumId: PropTypes.string,
        collectionIds: PropTypes.string
      })
    };

    static defaultProps = {
      refreshSession: refreshSession()
    };

    componentWillMount() {
      this.diffPropsAndRefresh(this.props);
    }

    componentWillReceiveProps(newProps) {
      this.diffPropsAndRefresh(newProps);
    }

    diffPropsAndRefresh(newProps) {
      const paramsDiffFromSession = !isEqualWith(
        newProps.params,
        newProps.appSession,
        (params, session) => {
          return (
            params.museumId * 1 === session.museumId &&
            params.collectionIds === session.collectionId
          );
        }
      );
      if (paramsDiffFromSession) {
        this.props.refreshSession(newProps.params, newProps.appSession);
      }
    }

    render() {
      return <Component {...this.props} />;
    }
  }
  Wrapper.displayName = `UrlAware(${getDisplayName(Component)})`;
  return Wrapper;
};

const initialState = { accessToken: getAccessToken() };

const loadAppSession = (ajaxGet = simpleGet, accessToken) => {
  accessToken = accessToken || getAccessToken();
  if (!accessToken) {
    return Observable.empty();
  }
  return Observable.forkJoin(
    ajaxGet(Config.magasin.urls.api.auth.buildInfo, accessToken),
    ajaxGet(Config.magasin.urls.api.actor.currentUser, accessToken),
    ajaxGet(Config.magasin.urls.api.auth.museumsUrl, accessToken)
  ).switchMap(([buildInfoRes, currentUserRes, museumsRes]) =>
    ajaxGet(
      Config.magasin.urls.api.auth.groupsUrl(currentUserRes.response.dataportenUser),
      accessToken
    ).map(({ response }) => {
      if (!response) {
        throw new Error(I18n.t('musit.errorMainMessages.noGroups'));
      }
      const isGod = !!response.find(group => 10000 === group.permission);
      let groups;
      if (isGod) {
        groups = museumsRes.response
          .filter(museum => 10000 !== museum.id)
          .map(museum => ({
            ...museum,
            museumId: museum.id,
            museumName: museum.shortName,
            permission: 10000,
            collections: [
              {
                uuid: '00000000-0000-0000-0000-000000000000',
                name: 'All'
              }
            ]
          }));
      } else {
        groups = response.map(group => ({
          ...group,
          museumName: museumsRes.response.find(m => m.id === group.museumId).shortName
        }));
      }
      const orderedGroups = orderBy(groups, ['museumId'], ['desc']);
      const museumId = orderedGroups[0].museumId;
      const collectionId = orderedGroups[0].collections[0].uuid;
      return {
        accessToken,
        actor: currentUserRes.response,
        groups: orderedGroups,
        museumId,
        collectionId,
        buildInfo: buildInfoRes.response,
        language: {
          isEn: getLanguage() === 'en',
          isNo: getLanguage() === 'no'
        }
      };
    })
  );
};

export const loadAppSession$ = createAction('loadAppSession$').switchMap(loadAppSession);
export const setMuseumId$ = createAction('setMuseumId$');
export const setCollectionId$ = createAction('setCollectionId$');
export const setAccessToken$ = createAction('setAccessToken$');

export const refreshSession = (
  setMuseum = id => setMuseumId$.next(id),
  setCollection = id => setCollectionId$.next(id)
) => (params, appSession) => {
  const museumId = appSession.museumId;
  const museumIdFromParam = parseInt(params.museumId, 10);
  if (museumIdFromParam && museumIdFromParam !== museumId) {
    setMuseum(museumIdFromParam);
  }
  const collectionId = appSession.collectionId;
  const collectionIdFromParam = params.collectionIds;
  if (collectionIdFromParam && collectionIdFromParam !== collectionId) {
    setCollection(collectionIdFromParam);
  }
};

export const reducer$ = (actions, onError = emitError) =>
  Observable.merge(
    actions.setAccessToken$.map(accessToken => state => ({ ...state, accessToken })),
    actions.loadAppSession$
      .map(session => state => ({ ...state, ...session }))
      .catch(error => {
        onError(error);
        return Observable.of(state => ({ ...state, accessToken: null }));
      }),
    actions.setMuseumId$.map(museumId => state => ({ ...state, museumId })),
    actions.setCollectionId$.map(collectionId => state => ({ ...state, collectionId }))
  );

const session$ = (
  actions$ = { setMuseumId$, setCollectionId$, setAccessToken$, loadAppSession$ }
) => createStore('appSession', reducer$(actions$), Observable.of(initialState));

export default session$();
