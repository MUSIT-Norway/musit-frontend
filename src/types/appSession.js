/* @flow */

import type { Actor } from './actor';

export type AppSession = {
  museumId: number,
  collectionId: string,
  accessToken: string,
  actor: Actor
};
