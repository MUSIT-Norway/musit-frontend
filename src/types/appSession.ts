// @flow

import { Actor } from './actor';
import { Uuid } from './common';

export type Language = {
  isEn: boolean;
  isNo: boolean;
};

interface BuildInfo {
  commitSha: string; //Eksempel: '813e993f1d1bdaa7b60daa66d75e1957de4c6e2b'
}

interface Collection {
  uuid: Uuid;
  name: string;
}
export interface Group {
  id: number;
  shortName: string;
  museumId: number;
  museumName: string;
  permission: number;
  collections: Array<Collection>;
}

export interface AppSession {
  museumId: number;
  collectionId: string;
  accessToken: string;
  actor: Actor;
  language: Language;
  rolesForModules: {
    collectionManagementRead: boolean;
    collectionManagementWrite: boolean;
    storageFacilityRead: boolean;
    storageFacilityWrite: boolean;
    storageFacilityAdmin: boolean;
    documentArchiveRead: boolean;
    documentArchiveWrite: boolean;
  };
  buildInfo?: BuildInfo;
  groups: Array<Group>;
  isGod?: boolean;
}
