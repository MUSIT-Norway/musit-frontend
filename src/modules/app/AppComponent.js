import { I18n } from 'react-i18nify';
import 'react-select/dist/react-select.css';
import React, { Component, PropTypes } from 'react';
import { IndexLink, hashHistory } from 'react-router';
import { LinkContainer } from 'react-router-bootstrap';
import { Navbar, Nav, NavItem } from 'react-bootstrap';
import FontAwesome from 'react-fontawesome';
import { connect } from 'react-redux';
import MusitUserAccount from './UserAccount';
import './AppComponent.css';
import Logo from './musitLogo.png';
import inject from '../../rxjs/inject';
import LoginComponent from '../login/LoginComponent';
import {emitError} from '../../shared/errors';
import notifiable from './Notifyable';
import Loader from 'react-loader';
import { SET_COLLECTION, SET_MUSEUM } from '../../redux/sessionReducer';
import { actions } from '../app/appSession';
const { loadAppSession$, setMuseumId$, setCollectionId$ } = actions;
import {Â AppSession } from './appSession';

export class App extends Component {
  static propTypes = {
    children: PropTypes.object.isRequired,
    appSession: PropTypes.instanceOf(AppSession).isRequired,
    setMuseumId: PropTypes.func.isRequired,
    setCollectionId: PropTypes.func.isRequired,
    loadAppSession: PropTypes.func.isRequired,
    setMuseumIdInRedux: PropTypes.func.isRequired,
    setCollectionIdInRedux: PropTypes.func.isRequired,
    picks: PropTypes.object
  }

  constructor(props, context) {
    super(props, context);
    this.handleLanguage = this.handleLanguage.bind(this);
    this.handleLogout = this.handleLogout.bind(this);
    this.handleMuseumId = this.handleMuseumId.bind(this);
  }

  componentWillMount() {
    this.props.loadAppSession();
  }

  handleLogout() {
    fetch('/api/auth/rest/logout', {
      method: 'GET',
      headers: new Headers({
        Authorization: 'Bearer ' + this.props.appSession.getAccessToken()
      })
    }).then(response => {
      if (response.ok) {
        localStorage.removeItem('jwtToken');
        localStorage.removeItem('accessToken');
        window.location.replace('https://auth.dataporten.no/logout');
      }
    }).catch(error => emitError({ type: 'network', error}));
  }

  handleLanguage(l) {
    localStorage.setItem('language', l);
    window.location.reload(true);
  }

  handleMuseumId(mid, cid) {
    this.props.setMuseumId(mid);
    this.props.setMuseumIdInRedux(mid);
    this.props.setCollectionId(cid);
    this.props.setCollectionIdInRedux(cid);
    hashHistory.push('/magasin');
  }

  handleCollectionId(nid, cid) {
    this.props.setCollectionId(cid);
    this.props.setCollectionIdInRedux(cid);
    if (nid) {
      hashHistory.push(`/magasin/${nid}`);
    } else {
      hashHistory.push('/magasin');
    }
  }
  getFooterClass(){
    let returnClassName;
    if (window.location.href.toLowerCase().includes('test:')) {
      returnClassName = 'footer backgroundUTV';
    } else if (window.location.href.toLowerCase().includes('utv.uio.no:')) {
      returnClassName = 'footer backgroundUTV';
    } else if (window.location.href.toLowerCase().includes('test.uio.no:')){
      returnClassName = 'footer backgroundTEST';
    } else {
      returnClassName = 'footer version well';
    }
    return returnClassName;
  }
  render() {
    if (!this.props.appSession.getAccessToken()) {
      return (
        <LoginComponent />
      );
    }
    if (!this.props.appSession.getBuildNumber()) {
      return <Loader loaded={false} />;
    }
    return (
      <div>
        <Navbar fixedTop style={{ zIndex:1 }}>
          <Navbar.Header>
            <Navbar.Brand>
              <IndexLink to={'/about'} activeStyle={{ color: '#33e0ff' }}>
                <div className="brand">
                  <img height="40" alt="logo" src={Logo} />
                </div>
                <span>MUSIT</span>
              </IndexLink>
            </Navbar.Brand>
            <Navbar.Toggle />
          </Navbar.Header>
          <Navbar.Collapse>
            <Nav navbar>
              <LinkContainer to="/magasin">
                <NavItem>{ I18n.t('musit.texts.magazine') }</NavItem>
              </LinkContainer>
              <LinkContainer to="/reports">
                <NavItem>{ I18n.t('musit.reports.reports') }</NavItem>
              </LinkContainer>
            </Nav>
            <Nav pullRight>
              <MusitUserAccount
                actor={this.props.appSession.getActor()}
                groups={this.props.appSession.getGroups()}
                token={this.props.appSession.getAccessToken()}
                selectedMuseumId={this.props.appSession.getMuseumId()}
                selectedCollectionId={this.props.appSession.getCollectionId()}
                handleLogout={this.handleLogout}
                handleLanguage={this.handleLanguage}
                handleMuseumId={this.handleMuseumId}
                handleCollectionId={(uuid) => this.handleCollectionId(this.props.rootNode && this.props.rootNode.id, uuid)}
                rootNode={this.props.rootNode}
              />
            </Nav>
            <Nav pullRight>
              <LinkContainer to={'/search/objects'}>
                <NavItem><FontAwesome name="search" style={{ fontSize: '1.3em' }} /></NavItem>
              </LinkContainer>
            </Nav>
            <Nav pullRight>
              <LinkContainer to="/picklist/node">
                <NavItem><span className="icon icon-musitpicklistnode" />{' '}{this.props.picks.NODE.length}</NavItem>
              </LinkContainer>
              <LinkContainer to="/picklist/object">
                <NavItem><span className="icon icon-musitpicklistobject" />{' '}{this.props.picks.OBJECT.length}</NavItem>
              </LinkContainer>
            </Nav>
          </Navbar.Collapse>
        </Navbar>

        <div className="appContent">
          {this.props.children}
        </div>

        <footer className={this.getFooterClass()}>
          {'Build number: ' + this.props.appSession.getBuildNumber()}
        </footer>
      </div>
    );
  }
}

const data = {
  appSession$: { type: PropTypes.object.isRequired }
};

const commands = {
  loadAppSession$,
  setMuseumId$,
  setCollectionId$
};

const mapStateToProps = (state) => ({
  picks: state.picks
});

const mapDispatchToProps = (dispatch) => ({
  setMuseumIdInRedux: (museumId) => dispatch({ type: SET_MUSEUM, museumId }),
  setCollectionIdInRedux: (collectionId) => dispatch({ type: SET_COLLECTION, collectionId })
});

export default connect(mapStateToProps, mapDispatchToProps)(notifiable(inject(data, commands)(App)));
