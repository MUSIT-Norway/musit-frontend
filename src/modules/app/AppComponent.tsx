import { I18n } from 'react-i18nify';
import 'react-select/dist/react-select.css';
import * as React from 'react';
import { Component } from 'react';
import * as PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { LinkContainer } from 'react-router-bootstrap';
import { Nav, Navbar, NavItem } from 'react-bootstrap';
import * as FontAwesome from 'react-fontawesome';
import MusitUserAccount from './components/UserAccount';
import './AppComponent.css';
import * as Logo from './musitLogo.png';
import LoginComponent from '../login/LoginComponent';
import { emitError } from '../../shared/errors';
import * as Loader from 'react-loader';
import {
  loadAppSession$,
  setCollectionId$,
  setMuseumId$,
  setRolesForModules$,
  MusitMatch
} from '../../stores/appSession';
import { RxInjectLegacy as inject } from '../../shared/react-rxjs-patch';
import {
  clearNodes$ as clearNodePicklist$,
  clearObjects$ as clearObjectPicklist$
} from '../../stores/pickList';
import Config from '../../config';
import { backendVersion, frontendVersion, VersionInfo } from '../../build';
import featureToggles, { FeatureToggles } from '../../featureToggles';
import * as classnames from 'classnames';
import env from '../../shared/environment';
import { clear$ as clearSearchStore$ } from '../../search/searchStore';
import { AppSession } from '../../types/appSession';
import { PicklistData } from '../../types/picklist';
import { TODO } from '../../types/common';

const about = '/about';

interface AppComponentProps {
  children: object;
  appSession: AppSession;
  setMuseumId: Function;
  setCollectionId: Function;
  loadAppSession: Function;
  setRolesForModules: Function;
  pickList: PicklistData;
  goTo: Function;
  clearObjectPicklist: Function;
  clearNodePicklist: Function;
  featureToggles: FeatureToggles;
  clearSearchStore: Function;
  match: MusitMatch;

  rootNode?: TODO;
}

/* Old:
 static propTypes = {
    children: PropTypes.object.isRequired,
    appSession: PropTypes.object.isRequired,
    setMuseumId: PropTypes.func.isRequired,
    setCollectionId: PropTypes.func.isRequired,
    loadAppSession: PropTypes.func.isRequired,
    setRolesForModules: PropTypes.func.isRequired,
    pickList: PropTypes.object.isRequired,
    goTo: PropTypes.func.isRequired,
    clearObjectPicklist: PropTypes.func.isRequired,
    clearNodePicklist: PropTypes.func.isRequired,
    featureToggles: PropTypes.object.isRequired,
    clearSearchStore: PropTypes.func.isRequired
  };

*/
export class AppComponent extends Component<AppComponentProps> {
  constructor(props: AppComponentProps, context: TODO) {
    super(props, context);
    this.handleLanguage = this.handleLanguage.bind(this);
    this.handleLogout = this.handleLogout.bind(this);
    this.handleMuseumId = this.handleMuseumId.bind(this);
    this.handleCollectionId = this.handleCollectionId.bind(this);
  }

  componentWillMount() {
    this.props.loadAppSession();
  }

  handleLogout() {
    fetch('/api/auth/rest/logout', {
      method: 'GET',
      headers: new Headers({
        Authorization: 'Bearer ' + this.props.appSession.accessToken
      })
    })
      .then(response => {
        if (response.ok) {
          localStorage.removeItem('accessToken');
          window.location.replace('https://auth.dataporten.no/logout');
        }
      })
      .catch(error => emitError({ type: 'network', error }));
  }

  handleLanguage(l: TODO) {
    localStorage.setItem('language', l);
    window.location.reload(true);
  }

  setMuseumCollectionId(museumId: TODO, collectionId: TODO) {
    localStorage.setItem('museumId', museumId);
    localStorage.setItem('collectionId', collectionId);
  }

  handleMuseumId(museumId: TODO, collectionId: TODO) {
    this.props.setMuseumId(museumId);
    this.props.setCollectionId(collectionId);
    this.props.setRolesForModules({
      email: this.props.appSession.actor.dataportenUser,
      museumId: museumId,
      collectionId: collectionId,
      isGod: this.props.appSession.isGod
    });
    this.props.clearObjectPicklist();
    this.props.clearNodePicklist();
    this.props.clearSearchStore();
    this.setMuseumCollectionId(museumId, collectionId);
    this.props.goTo(about);
  }

  handleCollectionId(collectionId: TODO) {
    this.props.setCollectionId(collectionId);
    this.props.setRolesForModules({
      email: this.props.appSession.actor.dataportenUser,
      museumId: this.props.appSession.museumId,
      collectionId: collectionId,
      isGod: this.props.appSession.isGod
    });
    this.props.clearObjectPicklist();
    this.props.clearSearchStore();
    this.setMuseumCollectionId(this.props.appSession.museumId, collectionId);
    this.props.goTo(about);
  }

  getFooterClass() {
    return classnames('footer', 'navbar-fixed-bottom', {
      backgroundUTV: env === 'utv',
      backgroundTEST: env === 'test',
      'version well': env === 'prod'
    });
  }

  isSessionLoaded() {
    return !!this.props.appSession.buildInfo;
  }

  buildVersionLink(versionInfo: VersionInfo) {
    if (versionInfo.url) {
      return (
        <a target="_blank" rel="noopener noreferrer" href={versionInfo.url}>
          {versionInfo.sha}
        </a>
      );
    }
    return versionInfo.sha;
  }

  render() {
    if (!this.props.appSession.accessToken) {
      return <LoginComponent />;
    }

    if (!this.isSessionLoaded()) {
      return <Loader loaded={false} />;
    }
    return (
      <div>
        <Navbar fixedTop>
          <Navbar.Header>
            <Navbar.Brand>
              <Link to={about}>
                <div className="brand">
                  <img height="40" alt="logo" src={Logo} />
                </div>
                <span>MUSIT</span>
              </Link>
            </Navbar.Brand>
            <Navbar.Toggle />
          </Navbar.Header>
          <Navbar.Collapse>
            <Nav navbar>
              {this.props.appSession.rolesForModules.storageFacilityRead && (
                <LinkContainer
                  to={Config.magasin.urls.client.magasin.goToMagasin(
                    this.props.appSession
                  )}
                >
                  <NavItem>{I18n.t('musit.texts.magazine')}</NavItem>
                </LinkContainer>
              )}
              <LinkContainer
                to={Config.magasin.urls.client.report.goToReport(this.props.appSession)}
              >
                <NavItem>{I18n.t('musit.reports.reports')}</NavItem>
              </LinkContainer>
              {this.props.featureToggles.administrationPage && (
                <LinkContainer
                  to={Config.magasin.urls.client.administration.goToAdministration(
                    this.props.appSession
                  )}
                >
                  <NavItem>{I18n.t('musit.administration.administration')}</NavItem>
                </LinkContainer>
              )}
            </Nav>
            <Nav pullRight>
              <LinkContainer
                to={Config.magasin.urls.client.picklist.goToPicklistNodes(
                  this.props.appSession
                )}
              >
                <NavItem>
                  <span className="icon icon-musitpicklistnode" />{' '}
                  {this.props.pickList.nodes.length}
                </NavItem>
              </LinkContainer>
              <LinkContainer
                to={Config.magasin.urls.client.picklist.goToPicklistObjects(
                  this.props.appSession
                )}
              >
                <NavItem>
                  <span className="icon icon-musitpicklistobject" />{' '}
                  {this.props.pickList.objects.length}
                </NavItem>
              </LinkContainer>
              <LinkContainer
                to={Config.magasin.urls.client.searchObjects.goToSearchObjects(
                  this.props.appSession
                )}
              >
                <NavItem>
                  <FontAwesome name="search" style={{ fontSize: '1.3em', height: 25 }} />
                </NavItem>
              </LinkContainer>
              <NavItem role="presentation" style={{ margin: '-15px' }}>
                <MusitUserAccount
                  actor={this.props.appSession.actor}
                  groups={this.props.appSession.groups}
                  token={this.props.appSession.accessToken}
                  selectedMuseumId={this.props.appSession.museumId}
                  selectedCollectionId={this.props.appSession.collectionId}
                  handleLogout={this.handleLogout}
                  handleLanguage={this.handleLanguage}
                  handleMuseumId={this.handleMuseumId}
                  handleCollectionId={this.handleCollectionId}
                  //?rootNode={this.props.rootNode}
                />
              </NavItem>
            </Nav>
          </Navbar.Collapse>
        </Navbar>

        <div className="appContent">{this.props.children}</div>

        <footer className={this.getFooterClass()}>
          {'Backend build: '}
          {this.buildVersionLink(backendVersion(this.props.appSession.buildInfo))}
          {' - Client build: '}
          {this.buildVersionLink(frontendVersion())}
        </footer>
      </div>
    );
  }
}

const props = (props: AppComponentProps) => ({
  ...props,
  featureToggles: featureToggles
});

const data = {
  appSession$: { type: PropTypes.object.isRequired },
  pickList$: { type: PropTypes.object.isRequired }
};

const commands = {
  loadAppSession$,
  setMuseumId$,
  setCollectionId$,
  clearObjectPicklist$,
  clearNodePicklist$,
  clearSearchStore$,
  setRolesForModules$
};

export default inject(data, commands, props)(AppComponent);
