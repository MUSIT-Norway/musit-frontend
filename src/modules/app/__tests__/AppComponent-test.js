import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import sinon from 'sinon';
import {Â AppComponent } from '../AppComponent';
import { AppSession } from '../appSession';

describe('AppComponent', () => {
  it('remders', () => {
    const prepareSearch = sinon.spy();
    const setMuseumId = sinon.spy();
    const setCollectionId = sinon.spy();
    const loadAppSession = sinon.spy();
    const scanForNodeOrObject = sinon.spy();
    const goTo = sinon.spy();
    const clearBuffer = sinon.spy();
    const clearObjects = sinon.spy();
    const clearNodes = sinon.spy();
    const toggleEnabled = sinon.spy();
    const wrapper = shallow(
      <AppComponent
        appSession={new AppSession({
          accessToken: '1234',
          buildInfo: {
            buildInfoBuildNumber: '64'
          },
          groups: [],
          actor: {
            dataportenId: '1234',
            fn: 'Jarl'
          }
        })}
        scanner={{}}
        prepareSearch={prepareSearch}
        setMuseumId={setMuseumId}
        setCollectionId={setCollectionId}
        loadAppSession={loadAppSession}
        pickList={{
          nodes: [
            {
              marked: false,
              value: {},
              path: []
            }
          ],
          objects: [
            {
              marked: false,
              value: {},
              path: []
            },
            {
              marked: false,
              value: {},
              path: []
            }
          ]
        }}
        scanForNodeOrObject={scanForNodeOrObject}
        goTo={goTo}
        clearBuffer={clearBuffer}
        clearObjects={clearObjects}
        clearNodes={clearNodes}
        toggleEnabled={toggleEnabled}
      >
        <span>Yay</span>
      </AppComponent>
    );
    expect(loadAppSession.calledOnce).toBe(true);
    wrapper.find('.toggleScanner').simulate('click', { preventDefault: () => true});
    expect(toggleEnabled.calledOnce).toBe(true);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});