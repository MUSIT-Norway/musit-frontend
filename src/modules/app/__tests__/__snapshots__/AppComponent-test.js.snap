exports[`AppComponent remders 1`] = `
<div>
  <Uncontrolled(Navbar)
    bsStyle="default"
    fixedTop={true}
    style={
      Object {
        "zIndex": 1
      }
    }>
    <NavbarHeader>
      <NavbarBrand>
        <IndexLink
          activeStyle={
            Object {
              "color": "#33e0ff"
            }
          }
          to="/about">
          <div
            className="brand">
            <img
              alt="logo"
              height="40"
              src="test-file-stub" />
          </div>
          <span>
            MUSIT
          </span>
        </IndexLink>
      </NavbarBrand>
      <NavbarToggle />
    </NavbarHeader>
    <NavbarCollapse>
      <Nav
        bsClass="nav"
        justified={false}
        navbar={true}
        pullLeft={false}
        pullRight={false}
        stacked={false}>
        <LinkContainer
          action="push"
          onlyActiveOnIndex={false}
          to="/museum/99/collections/00000000-0000-0000-0000-000000000000/magasin">
          <NavItem
            active={false}
            disabled={false}>
            Magasin
          </NavItem>
        </LinkContainer>
        <NavDropdown
          id="analysis-dropdown"
          title="Analyse">
          <LinkContainer
            action="push"
            onlyActiveOnIndex={false}
            to="/museum/99/collections/00000000-0000-0000-0000-000000000000/analysis/add">
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              header={false}>
              Registrer analyse
            </MenuItem>
          </LinkContainer>
          <LinkContainer
            action="push"
            onlyActiveOnIndex={false}
            to="/museum/99/collections/00000000-0000-0000-0000-000000000000/analysis/sample/undefined/add">
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              header={false}>
              Registrer pr√∏veuttak
            </MenuItem>
          </LinkContainer>
        </NavDropdown>
        <LinkContainer
          action="push"
          onlyActiveOnIndex={false}
          to="/museum/99/collections/00000000-0000-0000-0000-000000000000/reports">
          <NavItem
            active={false}
            disabled={false}>
            Rapporter
          </NavItem>
        </LinkContainer>
        <LinkContainer
          action="push"
          onlyActiveOnIndex={false}
          to="/museum/99/collections/00000000-0000-0000-0000-000000000000/administration">
          <NavItem
            active={false}
            disabled={false}>
            Administrasjon
          </NavItem>
        </LinkContainer>
      </Nav>
      <Nav
        bsClass="nav"
        justified={false}
        pullLeft={false}
        pullRight={true}
        stacked={false}>
        <LinkContainer
          action="push"
          onlyActiveOnIndex={false}
          to="/museum/99/collections/00000000-0000-0000-0000-000000000000/picklist/nodes">
          <NavItem
            active={false}
            disabled={false}>
            <span
              className="icon icon-musitpicklistnode" />
             
            1
          </NavItem>
        </LinkContainer>
        <LinkContainer
          action="push"
          onlyActiveOnIndex={false}
          to="/museum/99/collections/00000000-0000-0000-0000-000000000000/picklist/objects">
          <NavItem
            active={false}
            disabled={false}>
            <span
              className="icon icon-musitpicklistobject" />
             
            2
          </NavItem>
        </LinkContainer>
        <LinkContainer
          action="push"
          onlyActiveOnIndex={false}
          to="/museum/99/collections/00000000-0000-0000-0000-000000000000/search/objects">
          <NavItem
            active={false}
            disabled={false}>
            <FontAwesome
              name="search"
              style={
                Object {
                  "fontSize": "1.3em",
                  "height": 25
                }
              } />
          </NavItem>
        </LinkContainer>
        <MusitUserAccount
          actor={
            Object {
              "dataportenId": "1234",
              "fn": "Jarl"
            }
          }
          groups={Array []}
          handleCollectionId={[Function bound handleCollectionId]}
          handleLanguage={[Function bound handleLanguage]}
          handleLogout={[Function bound handleLogout]}
          handleMuseumId={[Function bound handleMuseumId]}
          rootNode={undefined}
          selectedCollectionId="00000000-0000-0000-0000-000000000000"
          selectedMuseumId={99}
          token="1234" />
      </Nav>
    </NavbarCollapse>
  </Uncontrolled(Navbar)>
  <div
    className="appContent">
    <span>
      Yay
    </span>
  </div>
  <footer
    className="footer version well">
    Backend build: 
    <a
      href="https://gitlab.com/MUSIT-Norway/musit/commit/813e993f1d1bdaa7b60daa66d75e1957de4c6e2b"
      target="_blank">
      813e993f1d1bdaa7b60daa66d75e1957de4c6e2b
    </a>
     - Client build: 
    LOCAL_DEVELOPMENT
  </footer>
</div>
`;
